File: Roadmap/02 - FUNCIONES Y ALCANCE/typescript/qv1ko.ts
let a: number = 3;

main();
console.log("\nNumber of times a text was not printed: " + program("zip", "zap"));

function main() {

    func1();
    console.log(func2());
    func3(a, "\nParameterized and non-return function:\nGlobal variable A: ");
    console.log(func4(-4, "\nFunction with return and with parameters:\nGlobal variable A: "));
    
}

function func1() {
    var b = 3;
    console.log("\nFunction without return and without parameters:\nGlobal variable A: " + a + "\nLocal variable B:" + b);
}

function func2(): string {
    return "\nFunction with return and without parameters:\nGlobal variable A: " + a;
}

function func3(num: number, str: string) {
    console.log(str + num);
}

function func4(num: number, str: string): string {
    return str + a + "\nAbsolute number: " + Math.abs(num);
}

function program(zip: string, zap: string): number {

    console.log("\nProgram:")

    let count: number = 0;

    for (let i = 1; i <= 100; i++) {
        if (i % 3 == 0 && i % 5 == 0) {
            console.log(zip + zap);
        } else if (i % 3 == 0) {
            console.log(zip);
        } else if (i % 5 == 0) {
            console.log(zap);
        } else {
            count++;
        }
    }

    return count;

}

File: Roadmap/02 - FUNCIONES Y ALCANCE/python/qv1ko.py
a = 5

def main():
    func1()
    print(func2())
    func3(a, "\nParameterized and non-return function:\nGlobal variable A: ")
    print(func4(-4, "\nFunction with return and with parameters:\nGlobal variable A: "))

def func1():
    b = 4
    print(f"\nFunction without return and without parameters:\nGlobal variable A: {a}\nLocal variable B: {b}")

def func2():
    txt = "\nFunction with return and without parameters:\nGlobal variable A: " + str(a)
    return txt

def func3(num, txt):
    print(f"{txt}{num}")

def func4(num, txt):
    text = txt + str(a) + "\nAbsolute number: " + str(abs(num))
    return text

main()

print("\nProgram:")

def program(zip, zap):

    count = 0

    for i in range(1, 100):
        if i % 3 == 0 and i % 5 == 0:
            print(zip + zap)
        elif i % 3 == 0:
            print(zip)
        elif i % 5 == 0:
            print(zap)
        else:
            count += 1
    
    return count

print("\nNumber of times a text was not printed: " + str(program("zip", "zap")))
